tests/ConstantTest.php

```diff
@@ -1482,6 +1513,92 @@ class A {
                 false,
                 '8.1',
             ],
+            'returnValueofNonExistantConstant' => [
+                '<?php
+                    class Foo
+                    {
+                        public const BAR = ["bar"];
+
+                        /**
+                         * @return value-of<self::BAT>
+                         */
+                        public function bar(): string
+                        {
+                            return self::BAR[0];
+                        }
+                    }
+                ',
+                'error_message' => 'UnresolvableConstant',
+            ],
+            'returnValueofStaticConstant' => [
+                '<?php
+                    class Foo
+                    {
+                        public const BAR = ["bar"];
+
+                        /**
+                         * @return value-of<static::BAR>
+                         */
+                        public function bar(): string
+                        {
+                            return static::BAR[0];
+                        }
+                    }
+                ',
+                'error_message' => 'UnresolvableConstant',
```

Comment:
Should this really be `UnresolvableConstant`? I think I'd find it confusing.

Reply:
> Should this really be `UnresolvableConstant`? I think I'd find it confusing.

Yeah, perhaps a bit confusing, but I made a note of it in the `UnresolvableConstant` documentation. I didn't really want to have another issue for `IllegalConstant` or something like that. I'm happy to change the name if you can come up with something that intuitively encompasses both usages, or would you like to have a separate issue?

Reply:
Maybe `UnresolvableLiteralConstant`?